pkg/sentry/fs/g3doc/fuse.md

```diff
@@ -141,6 +141,57 @@ ops can be implemented in parallel.
 -   a channel that the kernel task will be blocked on when the fd is not
     available.
 
+#### Basic I/O Implementation
+
+Currently we have implemented basic functionalities of read and write
+for our FUSE. We describe the design and ways to improve it here:
+
+##### Basic FUSE Read
+
+The vfs2 expects implementations of `vfs.FileDescriptionImpl.Read()` and
+`vfs.FileDescriptionImpl.PRead()`. When a syscall is made, it will eventually
+reach our implementation of those interface functions located at
+`pkg/sentry/fsimpl/fuse/regular_file.go` for regular files.
+
+After validation checks of the input, sentry sends `FUSE_READ` requests
+to the FUSE daemon. The FUSE daemon returns data after the `fuse_out_header`
+as the responses. Since there is no existing way to directly copy data from
```

Comment:
This isn't true, we could map memory from one process to another if we wanted to. But in this case, we don't want to, to keep the two processes isolated.